src/main/java/uk/gov/crowncommercial/dts/scale/cat/config/BuyerUserScheduler.java

```diff
@@ -0,0 +1,143 @@
+  @Scheduled(fixedDelayString = "PT1M")
+  public void scheduleSelfServiceBuyers() {
+    log.info("- Scheduler started to get Self Service Buyers -");
+    var buyersInJaggaer = this.getSelfServiceBuyers();
+
+    if (buyersInJaggaer == null || !"0".equals(buyersInJaggaer.getReturnCode())
+        || !"OK".equals(buyersInJaggaer.getReturnMessage())) {
+      log.info("Invalid response in scheduler");
+    } else if (buyersInJaggaer.getReturnCompanyData().size() != 1) {
+      log.info("Empty details in User-pool scheduler");
+    } else {
+      var buyersInTender = buyerDetailsRepo.findAll();
+      var usersToAdd = new ArrayList<SubUser>();
+
+      buyersInJaggaer.getReturnCompanyData().stream()
+          .forEach(e -> usersToAdd.addAll(e.getReturnSubUser().getSubUsers().stream()
+              .filter(ssoDataUser -> ssoDataUser.getSsoCodeData() != null)
+              .filter(subUser -> buyersInTender.stream()
+                  .noneMatch(supplier -> supplier.getUserId().equals(subUser.getUserId())))
+              .collect(Collectors.toList())));
+
+      log.info("Scheduler - Save unmatched Self Service Buyers in DB - userCount {}",
+          usersToAdd.size());
+      var savedUsers = buyerDetailsRepo.saveAll(usersToAdd.stream()
+          .map(e -> BuyerUserDetails.builder().userId(e.getUserId())
+              .userPassword(RandomStringUtils.randomAlphanumeric(8)).createdAt(Instant.now())
+              .createdBy("Scheduler").build())
+          .collect(Collectors.toList()));
+
+      generateExcelSheet(savedUsers);
+    }
+  }
+
+  private void generateExcelSheet(List<BuyerUserDetails> usersList) {
+    workbook = new XSSFWorkbook();
+    sheet = workbook.createSheet(SHEET_NAME);
+    Row row = sheet.createRow(0);
+    CellStyle style = workbook.createCellStyle();
+    XSSFFont font = workbook.createFont();
+    font.setBold(true);
+    font.setFontHeight(16);
+    style.setFont(font);
+    createCell(row, 0, "UserId", style);
+    createCell(row, 1, "Password", style);
+    writeData(usersList);
+  }
+
+  private void createCell(Row row, int columnCount, Object value, CellStyle style) {
+    sheet.autoSizeColumn(columnCount);
+    Cell cell = row.createCell(columnCount);
+    if (value instanceof Integer)
+      cell.setCellValue((Integer) value);
+    else
+      cell.setCellValue((String) value);
+
+    cell.setCellStyle(style);
+  }
+
+  private void writeData(List<BuyerUserDetails> usersList) {
+    CellStyle style = workbook.createCellStyle();
+    XSSFFont font = workbook.createFont();
+    font.setFontHeight(14);
+    style.setFont(font);
+    usersList.stream().forEach(user -> {
+      int rowCount = 1;
+      Row row = sheet.createRow(rowCount++);
+      int columnCount = 0;
+      createCell(row, columnCount++, user.getUserId(), style);
+      createCell(row, columnCount++, user.getUserPassword(), style);
+    });
+  }
+
+  private GetCompanyDataResponse getSelfServiceBuyers() {
```

Comment:
This is a bit troubling - we shouldn't have to reconfigure the Jaggaer web client.  I know you mentioned an auth issue with it, would be good to revisit when we can and figure out what's going on.  I shall try to have a look 👍

Reply:
@venkibric - did you have a chance to revisit this?  There is some specific config for the Jaggaer web client that we don't want to miss or duplicate - if we could debug and find out why the normal autowired client was not working that would be great 👍